tests/tst-semaphore.c

```diff
@@ -55,5 +56,32 @@ int main(void) {
     assert(sem_destroy(&sem_sync) == 0);
     assert(sem_destroy(&sem_done) == 0);
 
+    ///Named sempahore test
+
+    //Create and open two handles to a named semaphore
+    sem_t *named_sem1 = sem_open("name", O_CREAT, 0777, 1);
```

Comment:
Did you try this test on Linux too? If I read the sem_overview(7) manual page correctly, the name passed to sem_open() must start with a slash and have no other slashes. But what happens when it doesn't start with a slash? Please try to run this test on Linux.

Reply:
From the [documentation](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sem_open.html) I have read, it appears that the functionality of `sem_open` when given a name that does not begin with a slash is implementation dependent, as is the meaning of any non-leading slash. In the Linux environment that I am running this in, it passes. However, it does fail when adding any extra non-leading slash, which currently would pass with my implementation. Admittedly, the documentation does say that the name argument should adhere to the rules of a standard pathname, which (to me) would imply that leading slash. But, then it says that a name given without a leading slash causes implementation defined functionality. That extra bit seems to give free reign to do whatever when there is no leading slash, so I guess it is a little unclear. If you would like, I could add in this slash checking (leading, non-leading, both or either) functionality to align more with the Linux documentation.

Reply:
You're right that the error-case is less important than correctly supporting the correct case (with a leading slash which is the only slash), and no reasonable application would depend on the error case.